doc/design/program.md

```diff
@@ -15,7 +15,64 @@ optimize(cost)
 train(cost, reader=mnist.train())
 ```
 
-The first five lines of the following PaddlePaddle program generates, or, compiles, the `ProgramDesc` message.  The last line runs it.
+The first five lines of the following PaddlePaddle program generates,
+or, compiles, the `ProgramDesc` message. The last line runs it by
+generating the `ExecutionPlan` and sending to `Executor` for
+execution.
+
+
+### ProgramDesc
+
+The `ProgramDesc` describes the computation specified by the user, it
+will be the same regardless which devices the program runs on
+(CPU/single GPU/multiple GPU/multiple nodes), with the following
+requirements:
+
+1. It should be programming language agnostic. Currently, we have a
```

Comment:
should there be a way of exporting ProgramDesc? so that user can share it, like `export(cost, SAVE_TO_PATH)`? how we are going to differentiate saving algorithm(ProgramDesc)  from saving model?

Reply:
I think a model should be saved separately: `ProgramDesc` and the weights. So that the weights can be re-used for different `ProgramDescs`.
Maybe saving model is not strictly related to this PR, we can discuss more in a separate issue if we wish :)

Reply:
agree, thanks:)